underfs/hdfs/src/main/java/tachyon/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -294,10 +292,8 @@ public boolean isFile(String path) throws IOException {
       String[] rtn = new String[files.length];
       int i = 0;
       for (FileStatus status : files) {
-        TachyonURI filePathURI = new TachyonURI(status.getPath().toUri().toString());
-        String filePath =  NetworkAddressUtils.replaceHostName(filePathURI).toString();
         // only return the relative path, to keep consistent with java.io.File.list()
-        rtn[i ++] = filePath.substring(path.length()); // mUfsPrefix
+        rtn[i ++] =  status.getPath().getName();
```

Comment:
It still returns relative path, and I think it's safer than the substring approach which is pretty fragile and can lead to the issue mentioned on the ticket.